src/mono/mono/metadata/threads.c

```diff
@@ -4947,7 +4935,7 @@ ves_icall_System_Threading_Thread_StartInternal (MonoThreadObjectHandle thread_h
 	// HACK: threadpool threads can return to the JS event loop
 	// WISH: support this for other threads, too
```

Comment:
Maybe this should be TODO or FIXME instead of wish so it's easier to find in codebase audits. I assume we're going to get to it eventually either way though

Reply:
I should probably just add a managed `internal` flag on `MonoThreadObject` and get rid of this hack entirely.  I've more or less convinced myself that we will eventually need a wasm-specific API to set something like this on threads created with `var t = new Thread(...); t.Start()`, anyway